src/coreclr/jit/lsraxarch.cpp

```diff
@@ -2178,7 +2178,26 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
                 // DIV implicitly put op1(lower) to EAX and op2(upper) to EDX
                 srcCount += BuildOperandUses(op1, RBM_EAX);
                 srcCount += BuildOperandUses(op2, RBM_EDX);
-                srcCount += op3->isContained() ? BuildOperandUses(op3) : BuildDelayFreeUses(op3, op1);
+
+                if (!op3->isContained())
+                {
+                    // For non-contained nodes, we want to make sure we delay free the register for
+                    // op3 with respect to both op1 and op2. In other words, op3 shouldn't get same
+                    // register that is assigned to either of op1 and op2.
+
+                    RefPosition* op3RefPosition;
+                    srcCount += BuildDelayFreeUses(op3, op1, RBM_NONE, &op3RefPosition);
+                    if ((op3RefPosition != nullptr) && !op3RefPosition->delayRegFree)
```

Comment:
We don't need to also set it as delay free for `op3` because we don't track `delayFree` with regards to a specific operand, is that right?

Reply:
Correct. We just want to make sure that `op3` is marked as `delayFree` so neither of `op1` or `op2`'s registers conflict with that of `op3`.